src/utilstrencodings.cpp

```diff
@@ -227,8 +227,8 @@ std::vector<unsigned char> DecodeBase64(const char* p, bool* pfInvalid)
 
 std::string DecodeBase64(const std::string& str)
 {
-    vector<unsigned char> vchRet = DecodeBase64(str.c_str());
-    return (vchRet.size() == 0) ? string() : string((const char*)vchRet.data(), vchRet.size());
+    std::vector<unsigned char> vchRet = DecodeBase64(str.c_str());
```

Comment:
In commit "Remove uneccessary branches in utilstrencodings string constructors."

Some of the changes in this commit should be squashed into previous commit "In commit "Fix 0-subscript in utilstrencodings.cpp."

Also missing an n in unnecessary.